### PR TITLE
sumo/gcc 12.1 upgrade

### PR DESCRIPTION
**Pull request by jhershbe**
Update to gcc 12.1 and add gstreamer + ffmpeg to the feed

### CODE REVIEW

File: recipes-connectivity/ti-wifi-utils/ti-wifi-utils/0001-Fix-build-errors-with-gcc-10.patch
Comment (amstewart) [inline recipes-connectivity/ti-wifi-utils/ti-wifi-utils/0001-Fix-build-errors-with-gcc-10.patch:28]: Why is it appropriate to ignore this warning?

File: recipes-connectivity/ti-wifi-utils/ti-wifi-utils/0001-Fix-build-errors-with-gcc-10.patch
Comment (amstewart) [inline recipes-connectivity/ti-wifi-utils/ti-wifi-utils/0001-Fix-build-errors-with-gcc-10.patch]: Could you [add](https://github.com/ni/nilrt/blob/f882870a6aa2889466316e97aa72a91aa7bd33b8/docs/CONTRIBUTING.md#patch-files) an `Upstream-Status` trailer here?

File: recipes-connectivity/ti-wifi-utils/ti-wifi-utils/0001-Fix-build-errors-with-gcc-10.patch
Comment (amstewart) [inline recipes-connectivity/ti-wifi-utils/ti-wifi-utils/0001-Fix-build-errors-with-gcc-10.patch:40]: Could you describe why this change is appropriate in the `.patch` description?

File: conf/local.conf.sample
Comment (amstewart) [inline conf/local.conf.sample]: Enabling `commercial` licenses like this would whitelist all recipes with commercial licenses - which we definitely don't want.

Does something like `LICENSE_FLAGS_WHITELIST = "commercial_ffmpeg"` work instead, [per the manual](https://docs.yoctoproject.org/3.1.19/dev-manual/dev-manual-common-tasks.html#enabling-commercially-licensed-recipes)?

File: recipes-core/packagegroups/packagegroup-ni-extra.bb
Comment (amstewart) [inline recipes-core/packagegroups/packagegroup-ni-extra.bb]: Can the `ffmpeg` and `gstreamer` changes be broken out into their own patchset? I think we need some more time to figure out if we *can* distribute their content, and it doesn't seem to be blocking for the rest of the `gcc 12` patchsets.

**Review by amstewart — CHANGES_REQUESTED** [review]

File: recipes-connectivity/ti-wifi-utils/ti-wifi-utils/0001-Fix-build-errors-with-gcc-10.patch
Comment (jhershbe) [reply recipes-connectivity/ti-wifi-utils/ti-wifi-utils/0001-Fix-build-errors-with-gcc-10.patch:28]: Commented in the patch

File: recipes-connectivity/ti-wifi-utils/ti-wifi-utils/0001-Fix-build-errors-with-gcc-10.patch
Comment (jhershbe) [reply recipes-connectivity/ti-wifi-utils/ti-wifi-utils/0001-Fix-build-errors-with-gcc-10.patch]: Added

File: recipes-connectivity/ti-wifi-utils/ti-wifi-utils/0001-Fix-build-errors-with-gcc-10.patch
Comment (jhershbe) [reply recipes-connectivity/ti-wifi-utils/ti-wifi-utils/0001-Fix-build-errors-with-gcc-10.patch:40]: Described

File: conf/local.conf.sample
Comment (jhershbe) [reply conf/local.conf.sample]: Changed

File: recipes-core/packagegroups/packagegroup-ni-extra.bb
Comment (jhershbe) [reply recipes-core/packagegroups/packagegroup-ni-extra.bb]: Split out

**Review by amstewart — APPROVED** [review]

**Comment by amstewart** [comment]
Looks good to me. I'll hold off on pulling this until the oe-core PR is merged.

**Pull request merged**
